File: Problem 76.py
memoize = {}
def countSums(number, currMax=None):
    if number==0:
        return 1
    if currMax is None:
        currMax = number-1
    if (number, currMax) not in memoize:
        memoize[(number, currMax)] = 0
        for x in range(1, min(number, currMax)+1):
            memoize[(number, currMax)] += countSums(number-x, min(x,number-x))
    return memoize[(number, currMax)]
print(countSums(100))

File: Problem 76.hs
import Data.Function.Memoize (memoFix2)

countSums :: Integer -> Integer
countSums n = memoFix2 countSums' n $ n-1

countSums' :: (Integer -> Integer -> Integer) -> Integer -> Integer -> Integer
countSums' _ 0 _ = 1
countSums' f n m = sum $ map (\x -> f (n-x) $ min x $ n-x) [1.. min n m]

main = print $ countSums 100